dae/dae/genomic_resources/score_statistics.py

```diff
@@ -127,6 +129,131 @@ def add_value(self, value):
+        for hist_conf in histogram_desc:
+            has_min_max = "min" in hist_conf and "max" in hist_conf
+            if not has_min_max:
+                scr_min, scr_max = self._score_min_max(
+                    score, hist_conf["score"])
+                if "min" not in hist_conf:
+                    hist_conf["min"] = scr_min
+                if "max" not in hist_conf:
+                    hist_conf["max"] = scr_max
+            hist_configs[hist_conf["score"]] = hist_conf
+
+        chrom_histograms = []
+        with ProcessPoolExecutor(max_workers=self.jobs) as executor:
+            futures = []
+            chromosomes = score.get_all_chromosomes()
+            for chrom in chromosomes:
+                futures.append(executor.submit(self._do_hist,
+                                               chrom, hist_configs))
+            for future in futures:
+                chrom_histograms.append(future.result())
+
+        return self._merge_histograms(chrom_histograms)
+
+    def _do_hist(self, chrom, hist_configs):
+        histograms = {}
+        for scr_id, config in hist_configs.items():
+            histograms[scr_id] = Histogram.from_config(config)
+        score_names = list(histograms.keys())
+
+        score = open_score_from_resource(self.resource)
+        score_to_values = \
+            score.fetch_region(chrom, None, None, score_names)
+        for scr_id, vals_iter in score_to_values.items():
+            hist = histograms[scr_id]
+            for v in vals_iter:
+                if v is not None:  # None designates missing values
+                    hist.add_value(v)
+
+        return histograms
+
+    @staticmethod
+    def _merge_histograms(chrom_histograms):
+        res = {}
+        for histograms in chrom_histograms:
+            for scr_id, histogram in histograms.items():
+                if scr_id not in res:
+                    res[scr_id] = histogram
+                else:
+                    res[scr_id] = Histogram.merge(res[scr_id], histogram)
+        return res
+
+    def _score_min_max(self, score, score_id):
+        logger.info(f"Calculating min max for {score_id}")
+
+        chromosomes = score.get_all_chromosomes()
+        min_maxes = []
+        with ProcessPoolExecutor(max_workers=self.jobs) as executor:
+            futures = []
+            for chrom in chromosomes:
+                futures.append(executor.submit(self._min_max_for_chrom,
+                                               chrom, score_id))
+            for future in futures:
+                min_maxes.append(future.result())
+
+        limits = np.iinfo(np.int64)
+        scr_min, scr_max = limits.max, limits.min
+        for i_min, i_max in min_maxes:
+            scr_min = min(scr_min, i_min)
+            scr_max = max(scr_max, i_max)
+
+        logger.info(f"Done calculating min/max for {score_id}.\
+ min={scr_min} max={scr_max}")
+        assert scr_min <= scr_max
+        return scr_min, scr_max
```

Comment:
consider re-organizing min/max calculation to become similar to histogram calculation _do_hist/_merge_hist

consider keeping the parallelization into one place - inside build() method.

Reply:
@lchorbadjiev fixed